src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -110,6 +129,16 @@ public ODataSerializerContext(ResourceContext resource, SelectExpandClause selec
         /// </summary>
         public SelectExpandClause SelectExpandClause { get; set; }
 
+        /// <summary>
+        /// Gets or sets the <see cref="ExpandedNavigationSelectItem"/>.
+        /// </summary>
+        public ExpandedReferenceSelectItem ExpandedSelectItem { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ODataQueryOptions"/>.
+        /// </summary>
+        public ODataQueryOptions QueryOptions { get; set; }
```

Comment:
should have logic that sets SelectExpandClause when QueryOptions is set. Then we should deprecate SelectExpandClause in favor of QueryOptions.

Reply:
I have implemented something similar but I did not do it exactly as suggested to avoid a breaking change. 

---
In reply to: [291349027](https://github.com/OData/WebApi/pull/1724#discussion_r291349027) [](ancestors = 291349027)